### ISSUE TITLE
autocompletion not working on windows bash & powershell

### ISSUE THREAD

**Issue by patricklx** [spf13/cobra#1853]
its possible to remove the .exe extension to use binaries in windows git bash by specifying shopt -s completion_strip_exe
but bash still calls it with the exe extension. this leads to the issue that the completions are only registerd for the `binary.exe` and not also for just `binary`

e.g.: https://github.com/containers/podman/issues/16499

it registers only podman.exe: complete -o default -F __start_podman.exe podman.exe
it should also inlclude without .exe extension: complete -o default -F __start_podman.exe podman.exe podman.

**Comment by patricklx** [comment]
autocompletion for powerhsell also does not work as its using e.g `podman.exe` in the varaible names, turning it into a property lookup

**Comment by rhatdan** [comment]
We still want this one.

**Comment by Aetylus** [comment]
I'm unsure of how to fix the Powershell completer to register both to the `.exe` and non-`.exe` variant of the command in the template, but as far as getting the generated script to work, would it be safer to just always wrap potentially problematic names in curly braces `{}`?

For example, based off `podman.exe`, there are issues with the following lines in the Powershell template:
```
[scriptblock]$__%[2]sCompleterBlock
```
```
    # First disable ActiveHelp which is not supported for Powershell
    $env:%[10]s=0
```
```
Register-ArgumentCompleter -CommandName '%[1]s' -ScriptBlock $__%[2]sCompleterBlock
```

For Podman, these end up creating the following:

```
[scriptblock]$__podman.exeCompleterBlock = {
```
```
    # First disable ActiveHelp which is not supported for Powershell
    $env:PODMAN.EXE_ACTIVE_HELP=0
```
```
Register-ArgumentCompleter -CommandName 'podman','podman.exe' -ScriptBlock $__podman.exeCompleterBlock
```

All of these have problems due to the period making Powershell interpret them as properties. However, the following would work:
```
[scriptblock]${__podman.exeCompleterBlock} = {
```
```
    # First disable ActiveHelp which is not supported for Powershell
    ${env:PODMAN.EXE_ACTIVE_HELP}=0
```
```
Register-ArgumentCompleter -CommandName 'podman','podman.exe' -ScriptBlock ${__podman.exeCompleterBlock}
```

**Comment by mattcargile** [comment]
I've been having issues auto completing paths for exes like `glow` that use _cobra_ so I stumbled across this issue. The solution I have used is to add the below to my `$PROFILE` in `pwsh`. It appears this isn't an issue on `glow` at this time though.

```powershell
New-Alias -Name 'git.exe' -Value 'git' # Hack for autocomplete.
```

**Comment by marckhouzam** [comment]
@patricklx The powershell issue should be fixed by #1960
As for bash, I think the user should treat the `podman.exe` case as they need to treat aliases, and, if they use `podman` without the `.exe` extension, they will need to manually execute (or in their `.bashrc` file):
`complete -o default -F __start_podman.exe podman`
